### PR TITLE
Build container image in two steps (new action)

### PR DESCRIPTION
**Pull request by gvegayon**
This pull request introduces a new workflow for building and pushing container images in two steps, along with relevant documentation and examples. The most important changes include the addition of a new workflow file, updates to the action configuration, and new example files.

New workflow and action configuration:

* Added `test-twostep-container-build.yml` to define a new workflow for building and pushing container images in two steps.
* Created `twostep-container-build/action.yml` to configure the new action with necessary inputs and steps.

Documentation and examples:

* Updated `twostep-container-build/README.md` with detailed instructions and examples for using the new two-step container build action.
* Added example container files `twostep-container-build/examples/Containerfile` and `twostep-container-build/examples/Containerfile.dependencies` to demonstrate the two-step build process. [[1]](diffhunk://#diff-021ea60ff405e5c3ceb2a9bff2041bc47d2f4e8322bec3fee672d19d005e7274R1-R7) [[2]](diffhunk://#diff-9b307c78108548a91f90b60e04c2959d7684b6c2dddbe2c53d86bc52f88377e3R1-R5)
* Added `twostep-container-build/examples/example.yaml` as a sample workflow configuration for using the new action.

Minor changes:

* Removed `pull_request_target` trigger from `.github/workflows/test-post-artifact.yml`.

### CODE REVIEW

**Comment by gvegayon** [comment]
Hey @dylanhmorris, I've tagged you as a reviewer for this PR, but don't feel pressured! I'll see if we have someone else who could help with it (another maintainer for the cfa-actions project).

**Comment by gvegayon** [comment]
Hey @damonbayer! Could you take a look into this PR? This is relevant for pyrenew-hew, so it'd be great if we could merge this to update the workflow there.

File: twostep-container-build/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+| `password` | Password for the registry | true |  |
+| `registry` | Registry to push the image to | true |  |
+| `main-branch-name` | Name of the main branch | false | `'main'` |
+| `main-branch-tag` | Tag to use for the main branch | false | `'latest'` |
+| `push-image-1` | Push the image created during the first step | false | `false` |
+| `push-image-2` | Push the image created during the second step | false | `false` |
+
+## Example: Using ghcr.io
+
+The workflow is triggered on pull requests and pushes to the main branch. The image is pushed to `ghcr.io` and the image name is `cdcgov/cfa-actions` (full name is `ghcr.io/cdcgov/cfa-actions`). A functional version of this workflow is executed [here](../.github/workflows/test-twostep-container-build.yaml).
```
Comment (damonbayer) [inline twostep-container-build/README.md:32]:
```suggestion
The workflow is triggered on pull requests and pushes to the main branch. The image is pushed to `ghcr.io` and the image name is `cdcgov/cfa-actions` (full name is `ghcr.io/cdcgov/cfa-actions`). A functional version of this workflow is executed [here](../.github/workflows/test-twostep-container-build.yml).
```

**Review by damonbayer — APPROVED** [review]
@gvegayon I fixed a broken link. Otherwise, looks good to me! Thanks!

**Comment by gvegayon** [comment]
Thank you, @damonbayer! I'll create a PR to pyrenew-hew using this approach instead.

attn @zsusswein and @micahwiesner67. I'll do the same for the epinow2-pipeline and respnet-nowcasting repos.

**Pull request merged**